app/models/accounting/transaction.rb

```diff
@@ -131,19 +133,10 @@ def associate_with_qb_obj(qb_obj)
     self.quickbooks_data = qb_obj.as_json
   end
 
-  def change_in_principal
-    # TODO: Make project required and get rid of these guard clauses
-    return 0 unless project
-
-    # See InterestCalculator for more documentation on principal/interest accounts.
-    @change_in_principal ||= net_debit_for_account(qb_division&.principal_account_id)
-  end
-
-  def change_in_interest
-    return 0 unless project
-
-    # See InterestCalculator for more documentation on principal/interest accounts.
-     @change_in_interest ||= net_debit_for_account(qb_division&.interest_receivable_account_id)
+  def calculate_deltas
+    raise Accounting::TransactionMissingProjectError unless project
```

Comment:
This is a holdover from the comment above. Should we fix this special case here or in another issue?

Reply:
Another issue is fine, the last step of our bigger refactor should be a sweep of the code to find dead logic and inefficiencies, we can mop things up then.